staticd/static_vty.c

```diff
@@ -277,20 +308,20 @@ static int static_route_leak(struct vty *vty, const char *svrf,
 	} else {
 		if (src_str)
 			snprintf(ab_xpath, sizeof(ab_xpath),
-				 FRR_DEL_S_ROUTE_SRC_NH_KEY_XPATH,
+				 FRR_DEL_S_ROUTE_SRC_NH_KEY_NO_DISTANCE_XPATH,
 				 "frr-staticd:staticd", "staticd", svrf,
 				 buf_prefix,
 				 yang_afi_safi_value2identity(afi, safi),
-				 buf_src_prefix, table_id, distance,
-				 buf_nh_type, nh_svrf, buf_gate_str, ifname);
+				 buf_src_prefix, table_id, buf_nh_type, nh_svrf,
+				 buf_gate_str, ifname);
 		else
 			snprintf(ab_xpath, sizeof(ab_xpath),
-				 FRR_DEL_S_ROUTE_NH_KEY_XPATH,
+				 FRR_DEL_S_ROUTE_NH_KEY_NO_DISTANCE_XPATH,
 				 "frr-staticd:staticd", "staticd", svrf,
 				 buf_prefix,
 				 yang_afi_safi_value2identity(afi, safi),
-				 table_id, distance, buf_nh_type, nh_svrf,
-				 buf_gate_str, ifname);
+				 table_id, buf_nh_type, nh_svrf, buf_gate_str,
+				 ifname);
```

Comment:
just below here there's a "return ret" - but shouldn't that be WARNING_CONFIG_FAILED ?
(not your diff, I know, but...)

Reply:
You're right, but let's fix this in a separate PR.